pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java

```diff
@@ -398,32 +323,73 @@ private File buildFile(File base, String... pathParts) {
 	  StringBuilder path = new StringBuilder(base.getAbsolutePath());
 
 	  for (String part : pathParts) {
-		  path.append(File.separator).append(part);
+	      if(!StringUtils.isBlank(part)) {
+	          path.append(File.separator).append(part);
+	      }
 	  }
 
 	  return new File(path.toString());
   }
-
+  
   /**
-   * PIT timestamps reports to the minute which means it is possible to generate the same timestamped report twice.
-   * This function ensures that will not happen by waiting until the minute after the specified date time.
-   *
-   * @param startDateTime date time {@link String} in the format "yyyyMMddHHmm", this function will wait until a minute after this date time
-   * @throws Exception if this function waits more than 65 seconds or if there is an {@link InterruptedException} during the Thread.sleep
+   * Sets up a test of the reporting mojo simulating multiple runs of mvn install (as in the case where timestampedReports is set to true).  
+   * First calls {@link #prepareSiteTest(String)} then walks all directories starting at target/pit-reports setting their last modified time 
+   * to 0 (epoch time).  Finally, the directory specified in the {@code latestDir} parameter has its last modified time set to {@link System#currentTimeMillis()}.
+   * 
+   * @param testPath {@link String} see {@link #prepareSiteTest(String)}
+   * @param latestDir {@link String} containing the subdirectory of target/pit-reports that should be set as the latest, 
+   *                  pass an empty string to indicate the target/pit-reports directory should be the latest   
+   * 
+   * @return {@link File} representing the temporary folder that was set up for this execution of the test
+   * @throws Exception
    */
-  private void waitUntilNextMinute(String startDateTime) throws Exception {
-    LOGGER.info("wait for next minute");
-
-    //this code ensures that will not happen
-    int loopCount = 0;
-    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmm");
-    while(dateFormat.format(new Date()).equals(startDateTime)){
-    	if(loopCount > 65){
-    		throw new PitError("integration test is stuck in an infinite loop");
-    	}
-
-    	Thread.sleep(1000);
-    	loopCount++;
+  private File prepareSiteTest(String testPath, String latestDir) throws Exception {
+    File testDir = prepareSiteTest(testPath);
+    File testTargetDir = this.buildFile(testDir, "target", "pit-reports"); //location where the target directory would be if a mvn clean install was executed
+    DirectoriesOnlyWalker walker = new DirectoriesOnlyWalker();
+    
+    for(File f : walker.locateDirectories(testTargetDir)) {
+        f.setLastModified(0L);
     }
+    
+    buildFile(testTargetDir, latestDir).setLastModified(System.currentTimeMillis());
```

Comment:
Would you be so kind and rename the method `buildFile()` to `buildFilePath()` this describes more what it actually does. Thx!